master/buildbot/steps/trigger.py

```diff
@@ -181,11 +181,11 @@ def start(self):
             return
 
         was_exception = was_failure = False
-        brids = {}
+        brids = []
         for was_cb, results in rclist:
             if isinstance(results, tuple):
                 results, some_brids = results
```

Comment:
Problem is here some_brids is a dict with key is buildername, and value is buildrequestid.

So if you trigger several times the same builder, you got a problem.

I think it would be wise to rename a little bit the variables to make this more clear. There is too much magic in this code